docs/NEW_NEAR_CLI_INTERFACE.md

```diff
@@ -0,0 +1,114 @@
+# New NEAR CLI interface
+
+`NEAR CLI` is using `extensions` to saticfy all groups of users.
+- `Core NEAR CLI` commands should be usefull for all groups of users.
+- `Core NEAR CLI` is a single binary, there is no extensions that are installed `by default`.
+- Each extension is a separate binary that can be executed from `NEAR CLI`.
+- Extensions are used by a particular group or several groups of users.
+- Extensions are not composable (you should not create extensions for extensions).
+
+`NEAR CLI` is built for:
+- NEAR `dApp developers`, who build smart-contracts, UIs, and tooling on NEAR.
+- `Tech-savvy` people automating their routines.
+- `Validators`
+
+`NEAR CLI UX principles`
+- All altering actions should have a confirmation step with an option to skip confirmation with an explicit command line parameter (e.g. `send` at the end of the command)
+- All direct children commands of a single parent command should be aligned (either represent an action or a resource, but never a mix of those on the same hierarchy level): `contract` -> `state` (resource) and `contract` -> `deploy` (action) are not aligned, so it should be either `contract` -> `get-state` + `contract` -> `deploy` or `contract` -> `state` -> `view` + `contract` -> `code` -> `deploy`
+- Interactive mode should look like: `command - description`. It will help people to learn the commands.
+
+
+## `Core NEAR CLI` commands
+```txt
+import-account
+    - from-near-wallet
+    - from-seed-phrase <seed-phrase>
+    - from-ledger
+    - from-private-key <private-key>
+
+create-account
+    - implicit
+    - subaccount <master-account> <new-subaccount-id>
+
+manage-account <accoundId>
+    - get-state
+    - manage-keys
+        - view
+        - add
+        - delete <public-key>
+    - make-transfer <sender> <reciever> <amount>
+    - manage-contract
+        - deploy-code <wasm-path>
+        - get-code-checksum
+        - get-state
```

Comment:
How is it different from the `get-state` on the account level?

Reply:
Account state is a representation of this structure in `nearcore`:
```rust
pub struct Account {
    /// The total not locked tokens.
    #[serde(with = "u128_dec_format_compatible")]
    amount: Balance,
    /// The amount locked due to staking.
    #[serde(with = "u128_dec_format_compatible")]
    locked: Balance,
    /// Hash of the code stored in the storage for this account.
    code_hash: CryptoHash,
    /// Storage used by the given account, includes account id, this struct, access keys and other data.
    storage_usage: StorageUsage,
    /// Version of Account in re migrations and similar
    #[serde(default)]
    version: AccountVersion,
}
```

And here, the state is something similar to https://github.com/near-examples/near-account-utils#storage-report. Maybe it should be renamed.

Reply:
Ah, that is indeed not intuitive...